openhtf/io/output/__init__.py

```diff
@@ -12,15 +12,79 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Openhtf output modules."""
+"""This module contains support for various built-in output mechanisms.
 
-import openhtf.io.output.json_factory
-import openhtf.io.output.mfg_inspector
+Here, a base OutputToFile class is implemented to provide simple output to
+a file via the pickle serialization mechanism. It can be subclassed to implement
+alternative serialization schemes, see json_factory.py and mfg_inspector.py for
+examples.
+"""
 
-# pylint: disable=invalid-name
+import base64
+import contextlib
+import cPickle as pickle
+import os
+import tempfile
+from openhtf.util import data
 
-# Classes used for output
-OutputToJSON = json_factory.OutputToJSON
-UploadToMfgInspector = mfg_inspector.UploadToMfgInspector
-OutputToTestRunProto = mfg_inspector.OutputToTestRunProto
-UploadOrOutput = mfg_inspector.UploadOrOutput
+
+# TODO(wallacbe): Switch to util
+class Atomic(object):
+  """Class that does atmomic write in a contextual manner."""
+
+  def __init__(self, filename):
+    self.filename = filename
+    self.temp = tempfile.NamedTemporaryFile(delete=False)
+
+  def write(self, write_data):
+    self.temp.write(write_data)
+
+  def close(self):
+    self.temp.close()
+    os.rename(self.temp.name, self.filename)
+
+
+class OutputToFile(object):
+  """Class for formatting filename and writing to file."""
+
+  def __init__(self, filename_pattern):
+    self.filename_pattern = filename_pattern
+
+  @staticmethod
+  def serialize_test_record(test_record):
```

Comment:
can you throw a quick docstring on these describing why a subclass might override it?  so for this one, something like "Override this to alter how test records are serialize to file data."

and `open_file` might have "Override this to alter how output files are opened or what type of file is used."